spec.html

```diff
@@ -35961,15 +36216,33 @@ <h1>
         </emu-alg>
         <emu-grammar>Atom :: `.`</emu-grammar>
         <emu-alg>
-          1. Let _A_ be the CharSet of all characters.
+          1. Let _A_ be AllCharacters(_rer_).
           1. If _rer_.[[DotAll]] is not *true*, then
             1. Remove from _A_ all characters corresponding to a code point on the right-hand side of the |LineTerminator| production.
           1. Return CharacterSetMatcher(_rer_, _A_, *false*, _direction_).
         </emu-alg>
         <emu-grammar>Atom :: CharacterClass</emu-grammar>
         <emu-alg>
           1. Let _cc_ be CompileCharacterClass of |CharacterClass| with argument _rer_.
-          1. Return CharacterSetMatcher(_rer_, _cc_.[[CharSet]], _cc_.[[Invert]], _direction_).
+          1. Let _cs_ be _cc_.[[CharSet]].
+          1. If _rer_.[[UnicodeSets]] is *false*, or if every element of _cs_ consists of a single character, return CharacterSetMatcher(_rer_, _cs_, _cc_.[[Invert]], _direction_).
```

Comment:
A |CharacterClass| can be empty. I gather that's intended to fall under this branch, but it's not always obvious to readers when how things which are vacuously true should be interpreted, so I'd prefer to be explicit:

```suggestion
          1. If _rer_.[[UnicodeSets]] is *false*, or if every element of _cs_ consists of a single character (including if it is empty), return CharacterSetMatcher(_rer_, _cs_, _cc_.[[Invert]], _direction_).
```

Reply:
Done